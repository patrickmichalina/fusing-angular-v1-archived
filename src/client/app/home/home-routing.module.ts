import { HomeComponent } from './home.component'
import { NgModule } from '@angular/core'
import { RouterModule } from '@angular/router'

@NgModule({
  imports: [
    RouterModule.forChild([
      {
        path: '',
        component: HomeComponent
      }
    ])
  ],
  exports: [RouterModule]
})
export class HomeRoutingModule { }
