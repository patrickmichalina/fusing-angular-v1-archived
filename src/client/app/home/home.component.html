<h1>Fusing Angular</h1>
<h2>An Angular Universal seed project featuring Server-Side Rendering, @fuse-box bundling, Jest, Nightmare, and more!</h2>
<p>The goal of this project is to provide a developer friendly Angular Universal starter application. It differs from the Angular-CLI by focusing soley on Universal applications and uses faster testing frameworks and javascript bundler. With this project you can deploy a boilerplate application with a single click via Heroku.</p>
<div>
  <div>
    <a class="github-button" href="https://github.com/patrickmichalina/fusing-angular" data-icon="octicon-star" data-size="large"
      data-show-count="true" aria-label="Star patrickmichalina/fusing-angular on GitHub">Star</a>
    <a class="github-button" href="https://github.com/patrickmichalina/fusing-angular/archive/master.zip" data-icon="octicon-cloud-download"
      data-size="large" aria-label="Download patrickmichalina/fusing-angular on GitHub">Download</a>
  </div>
  <br>
  <a href="https://heroku.com/deploy?template=https://github.com/patrickmichalina/fusing-angular" target="_blank" rel="noopener">
    <img src="https://www.herokucdn.com/deploy/button.svg" alt="Deploy">
  </a>
</div>
<p>Features:</p>
<ul>
  <li>Angular as the application framework</li>
  <li>FuseBox as the TypeScript/JavaScript bundler</li>
  <li>Jest for unit and component testing</li>
  <li>Nightmare for E2E UI testing</li>
  <li>Sparky as the task runner</li>
  <li>Heroku Single Click Deployment</li>
  <li>SCSS support for professional grade CSS management</li>
  <li>Brotli compression with gzip fallback</li>
  <li>CircleCI integration</li>
  <li>Tree-Shaking for production builds</li>
  <li>Ahead-of-Time (AOT) compilation support</li>
  <li>angular-tslint-rules as configuration preset for TSLint and codelyzer.</li>
  <li>Lazy Loaded modules</li>
  <li>Fully typed build tools using TypeScript</li>
  <li>Manage your type definitions using @types</li>
  <li>Favicon icon generation for multiple devices derived from a single seed image</li>
  <li>Search Engine (SEO) support for Title and Meta tags</li>
  <li>OG (Open Graph) tags for social sharing</li>
  <li>Vendor-agnostic analytics using angulartics2</li>
  <li>Generic token based Authentication service with JWT cookie support.</li>
  <li>Analyze bundle sizes by using source-map-explorer</li>
  <li>Support for Angular Service Worker</li>
</ul>