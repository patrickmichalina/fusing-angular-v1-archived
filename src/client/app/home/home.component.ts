import {
  ChangeDetectionStrategy,
  Component,
  ViewEncapsulation
} from '@angular/core'

@Component({
  selector: 'pm-home',
  templateUrl: './home.component.html',
  styleUrls: ['./home.component.css'],
  changeDetection: ChangeDetectionStrategy.OnPush,
  encapsulation: ViewEncapsulation.None
})
export class HomeComponent {}
