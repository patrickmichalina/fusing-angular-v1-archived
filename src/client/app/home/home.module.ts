import { HomeRoutingModule } from './home-routing.module'
import { HomeComponent } from './home.component'
import { NgModule } from '@angular/core'
import { SharedModule } from '../shared/shared.module'
import { MarkdownToHtmlModule } from 'markdown-to-html-pipe'

@NgModule({
  imports: [HomeRoutingModule, SharedModule, MarkdownToHtmlModule],
  declarations: [HomeComponent],
  exports: [HomeComponent]
})
export class HomeModule {}
