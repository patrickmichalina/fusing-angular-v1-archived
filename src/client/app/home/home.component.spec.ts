import { HomeComponent } from './home.component'
import { async, ComponentFixture, TestBed } from '@angular/core/testing'
import { Component } from '@angular/core'
import { HomeModule } from './home.module'
import { AppTestingModule } from '../../../testing/app-testing.module'

describe(HomeComponent.name, () => {
  let fixture: ComponentFixture<HomeComponent>

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [AppTestingModule.forRoot(), HomeModule],
      declarations: [TestComponent]
    }).compileComponents()
  }))

  beforeEach(async(() => {
    fixture = TestBed.createComponent(HomeComponent)
  }))

  afterEach(async(() => {
    TestBed.resetTestingModule()
  }))

  it('should compile', async(() => {
    expect(fixture.nativeElement).toBeDefined()
    expect(fixture).toMatchSnapshot()
  }))
})

@Component({
  selector: 'test-component',
  template: '<pm-home></pm-home>'
})
class TestComponent { }
