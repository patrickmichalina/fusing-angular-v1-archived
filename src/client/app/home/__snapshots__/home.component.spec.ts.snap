// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeComponent should compile 1`] = `
<pm-home>
  Home
</pm-home>
`;
